src/sentry/integrations/gitlab/webhooks.py

```diff
@@ -193,12 +193,19 @@ def dispatch(self, request: Request, *args, **kwargs) -> Response:
         return super().dispatch(request, *args, **kwargs)
 
     def post(self, request: Request) -> Response:
+        extra = {
+            # This tells us the Gitlab version being used (e.g. current gitlab.com version -> GitLab/15.4.0-pre)
+            "user-agent": request.META.get("HTTP_USER_AGENT"),
+            "event-type": request.META["HTTP_X_GITLAB_EVENT"],
+        }
```

Comment:
`extra` will now be passed to all responses. I add metadata to this structure as soon as it's available.